umongo/data_proxy.py

```diff
@@ -85,16 +85,27 @@ def dump(self, schema=None):
     def _mark_as_modified(self, key):
         self._modified_data.add(key)
 
-    def update(self, data, schema=None):
+    def update(self, data, schema=None, reset_missings=False):
         schema = schema or self._schema
-        # Always use marshmallow partial load to skip required checks
-        loaded_data, err = schema.load(data, partial=True)
+
+        # Use marshmallow partial load to skip required checks
+        # unless reset_missings is True
+        loaded_data, err = schema.load(data, partial=(not reset_missings))
         if err:
             raise ValidationError(err)
         self._data.update(loaded_data)
         for key in loaded_data:
             self._mark_as_modified(key)
 
+        # Delete missing fields unless dump_only or with a default value
+        if reset_missings:
+            deletable_fields = set(
+                [k for k, v in schema.fields.items()
+                 if not v.dump_only and v.missing is missing])
```

Comment:
Not sure about the `v.missing is missing` filter...
I would say all field should be reset, either back to `missing` object or to their custom `field.missing` value (or to `field.missing()` if it's a callable)